### PR TITLE
Allow editors to setup the main menu, and add it in frontend DDFFORM-221

### PR DESCRIPTION
**Pull request by rasben**
Allow editors to setup the main menu, and add it in frontend DDFFORM-221

This includes a contrib module, for setting up permissions per
menu, so we avoid that the editors can edit other menus.

-----

Hide unused/unwanted functioanlity from Menu UI. DDFFORM-221

We want to avoid the editor setting up multi-level menus, as
the frontend does not support it.
We'll hide the options using CSS, but, if they do get around it,
it's not the end of the world as it just will have no effect.

-----

Setup main menu default content.

----

### CODE REVIEW

File: web/themes/custom/novel/templates/layout/menu--main.html.twig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
+<ul class="header__menu-navigation">
+{% for item in items %}
+  <li class="header__menu-navigation-item">
+    <a href="{{ item.url }}" class="header__menu-navigation-link text-body-medium-regular hide-linkstyle">
+      {{ item.title }}
+    </a>
```
Comment (kasperg) [inline web/themes/custom/novel/templates/layout/menu--main.html.twig:7]: Consider using the [`link()` function](https://www.drupal.org/docs/develop/theming-drupal/twig-in-drupal/functions-in-twig-templates#s-linktext-uri-attributes) and `item.attributes`.

Attributes is often helpful if we get to marking the current menu item as active. The link function also has helpers specifically regarding menu items that may be helpful.

File: web/modules/custom/dpl_example_content/dpl_example_content.info.yml [hunk 2]
```diff
@@ -56,3 +56,7 @@ default_content:
     - 0502836d-0f7b-4b21-9371-c40ae9e01495
     - 0fa7245a-3434-4d96-9793-a1848a22d37a
     - 807e6053-86ed-4dce-b6a9-a48b566b5910
+  menu_link_content:
+    - ff6db3ce-49e0-4107-b283-3403e22c5e6d
+    - c391079e-a4f5-4d29-992f-92ed1a2dd96c
+    - a25ebb74-e87f-4b2c-8156-b622dc23844e
```
Comment (kasperg) [inline web/modules/custom/dpl_example_content/dpl_example_content.info.yml:62]: Consider adding all the elements from Figma to the default content.

This makes it easy for us to replicate what is currently the business reference point.

File: web/modules/custom/dpl_example_content/content/menu_link_content/ff6db3ce-49e0-4107-b283-3403e22c5e6d.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      value: true
+  title:
+    -
+      value: 'Det sker'
+  menu_name:
+    -
+      value: main
+  link:
+    -
+      uri: 'internal:/'
```
Comment (kasperg) [inline web/modules/custom/dpl_example_content/content/menu_link_content/ff6db3ce-49e0-4107-b283-3403e22c5e6d.yml:19]: Consider linking to the `/events` page.

It is a reasonable assumption that this menu item would actually take the user there.

**Review by kasperg — APPROVED** [review]
:+1: LGTM.

I have a few comments for your consideration before you merge.

File: web/modules/custom/dpl_admin/assets/dpl_admin.css [hunk 4]
```diff
@@ -87,3 +87,16 @@
...
+.menu-link-content-form .field--name-description,
+.menu-link-content-form .field--name-expanded,
+.menu-link-content-form .form-item--menu-parent {
+  display: none;
+}
+
+.menu-edit-form[action="/admin/structure/menu/manage/main"] .tabledrag-handle,
+.menu-edit-form[action="/admin/structure/menu/manage/main"] > .form-item {
+  display: none;
+}
```
[the remark this answers is not in the document]
Comment (rasben) [reply web/modules/custom/dpl_admin/assets/dpl_admin.css:102]: Good point, i didnt consider that, i'll find another solution :)

[the remark this answers is not in the document]
Comment (rasben) [reply web/modules/custom/dpl_admin/assets/dpl_admin.css]: I've taken this out of the PR, and i'll introduce it in another.

File: web/themes/custom/novel/templates/layout/menu--main.html.twig (hunk 1 above)
Comment (rasben) [reply web/themes/custom/novel/templates/layout/menu--main.html.twig]: Good point, done 👍

File: web/modules/custom/dpl_example_content/dpl_example_content.info.yml (hunk 2 above)
Comment (rasben) [reply web/modules/custom/dpl_example_content/dpl_example_content.info.yml:62]: Good point, done 👍

File: web/modules/custom/dpl_example_content/content/menu_link_content/ff6db3ce-49e0-4107-b283-3403e22c5e6d.yml (hunk 3 above)
Comment (rasben) [reply web/modules/custom/dpl_example_content/content/menu_link_content/ff6db3ce-49e0-4107-b283-3403e22c5e6d.yml]: Good point, done 👍

**Pull request merged**